builtin/mainmenu/dlg_create_world.lua

```diff
@@ -17,13 +17,115 @@
 
 local worldname = ""
 
+local function flags_to_table(flags)
+	if flags then
+		local ftable = flags:gsub("%s+", ""):split(",", true) -- Remove all spaces and split
+		local ftable2 = {}
+		for f=1, #ftable do
+			if string.sub(ftable[f], 1, 2) == "no" then
+				ftable2[string.sub(ftable[f], 3)] = false
+			else
+				ftable2[ftable[f]] = true
+			end
+		end
+		return ftable2
+	else
+		return {}
+	end
+end
+
+local function table_to_flags(ftable)
+	local str = ""
+	local first = true
+	for flag, is_set in pairs(ftable) do
+		if not first then
+			str = str .. ","
+		end
+		if is_set then
+			str = str .. flag
+		else
+			str = str .. "no" .. flag
+		end
+		first = false
```

Comment:
You could use a table here.
1) Takes care of tailing comma separation
2) Is faster than string concat
3) Is shorter
```Lua
-- Convert { jungles = true, caves = false } to "jungles,nocaves"
local str = {}
for flag, is_set in pairs(ftable) do
	str[#str + 1] = is_set and flag or ("no" .. flag)
end
return table.concat(str, ",")